sdk/storage/Azure.Storage.Common.DataMovement/src/DirectoryScanner.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Azure.Storage
+{
+    internal class DirectoryScanner
+    {
+        private string _path;
+
+        public DirectoryScanner(string path)
+        {
+            _path = path;
+        }
+
+        // Constuctor for mocking
+        public DirectoryScanner() { }
```

Comment:
I saw that in ChangeFeed and its tests, the Segment, SegmentFactory, etc had these kinds of constructors so their outputs could be manipulated for mocking in the tests; I was thinking this could be for mocking specific file structures, so setting up maybe 3 DirectoryScanners, fixing the outputs, and the fixing the DirectoryScannerFactory output to these three scanners. I guess this is kind of unnecessary if we can just do temp files, though.

Reply:
In Changefeed I believe these were created for Moq (https://github.com/Moq/moq4/wiki/Quickstart ). But it turns out these are unnecessary. Moq can deal with parametrized ctors.

Reply:
Cool, I'll take these out across the refactor (and since I can move the methods in here to private methods in FileSystemScanner anyways if the mocking is needed, this class and its factory will disappear)